src/interfaces/hapi.ts

```diff
@@ -79,30 +79,46 @@ export function makeHapiPlugin(client: RequestHandler, config: Configuration) {
    * @returns {Undefined} - returns the execution of the next callback
    */
   function hapiRegisterFunction(server: {}, options: {}, next: Function) {
-    if (isObject(server)) {
-      if (isFunction((server as hapi.Server).on)) {
-        (server as hapi.Server).on('request-error', (req, err) => {
-          client.sendError(hapiErrorHandler(req, err, config));
+    if (server) {
+      if ((server as any).events && (server as any).events.on) {
```

Comment:
Do you need these `server as any` because `@types/hapi` doesn't support hapi 17 yet?

Reply:
We need to support hapi 16 and hapi 17 at the same time which is tricky from a types perspective.